runelite-client/src/main/java/net/runelite/client/plugins/statusorbs/StatusOrbsOverlay.java

```diff
@@ -222,4 +265,61 @@ private void renderRegen(Graphics2D g, WidgetInfo widgetInfo, double percent, Co
 		g.setColor(color);
 		g.draw(arc);
 	}
+
+	public void updateSpec()
+	{
+		//0.2% spec restore per tick
+		if (client.getItemContainer(InventoryID.EQUIPMENT) == null)
+		{
+			return;
+		}
+		currentSpec = Math.floor((client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT) / 10) + (plugin.getSpecialPercentage() * 10));
+		int specTarget = 100;
+		final Item[] items = client.getItemContainer(InventoryID.EQUIPMENT).getItems();
+		if (items.length <= EquipmentInventorySlot.WEAPON.getSlotIdx())
+		{
+			return;
+		}
+		final Item weapon = items[EquipmentInventorySlot.WEAPON.getSlotIdx()];
+		final ItemDefinition weaponComp = itemManager.getItemDefinition(weapon.getId());
+
+		String weaponName;
+		//magic shortbow(i) had to be special
+		if (weaponComp.getName().toLowerCase().equals("magic shortbow (i)"))
+		{
+			weaponName = "magic shortbow (i)";
+		}
+		else
+		{
+			weaponName = weaponComp.getName().replaceAll("\\([^()]*\\)", "").toLowerCase();
+		}
+
+		if (weapons.containsKey(weaponName))
+		{
+			if (weaponName.equals("granite maul") && Arrays.asList(24225, 24226, 24227, 24228).contains(weaponComp.getId()))
+			{
+				specTarget = 50;
+			}
+			else
+			{
+				specTarget = weapons.get(weaponName);
+			}
+		}
+		specCount = (int) (currentSpec / specTarget);
```

Comment:
I would personally go a entirely different way with how to calculate when next spec change is, pretty much [how the poison plugin does it](https://github.com/open-osrs/runelite/blob/d1b4e2e23f0892708baa6ce845d114d2e0e64242/runelite-client/src/main/java/net/runelite/client/plugins/poison/PoisonPlugin.java#L216). It could do with also counting client ticks between gameticks, to make the timer a little bit smoother but in the end that doesn't really make a huge difference (if at all).

Integer math and reusing values rather than recalculating every time would improve efficiency over this impl